File: 14/list-14.py
class Configuracao:
    _instancia = None 

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super().__new__(cls) 
            print("Nova instância criada.")
        return cls._instancia  

def main():
    config = Configuracao() 
    outra_config = Configuracao()
    print(f"É a mesma instância? {config is outra_config}")

if __name__ == "__main__":
    main()

File: 14/list-14.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct{}

var instancia *Configuracao
var once sync.Once

func GetConfiguracao() *Configuracao {
	once.Do(func() {
		fmt.Println("Nova instância criada.")
		instancia = &Configuracao{}
	})
	return instancia
}

func main() {
	config := GetConfiguracao()
	outraConfig := GetConfiguracao()

	fmt.Printf("É a mesma instância? %v\n", config == outraConfig)
}
